subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/maven/DefaultMavenModuleResolveMetadata.java

```diff
@@ -108,7 +108,7 @@ protected DefaultConfigurationMetadata createConfiguration(ModuleComponentIdenti
 
     @Override
     protected Optional<ImmutableList<? extends ConfigurationMetadata>> maybeDeriveVariants() {
-        return isJavaLibrary() ? Optional.<ImmutableList<? extends ConfigurationMetadata>>of(getDerivedVariants()) : Optional.<ImmutableList<? extends ConfigurationMetadata>>absent();
+        return Optional.<ImmutableList<? extends ConfigurationMetadata>>of(getDerivedVariants());
```

Comment:
:x: Can now remove `isJavaLibrary()`
:x: Need to make the same change in `RealizedMavenModuleResolveMetadata`